app/src/main/java/com/kodeco/android/countryinfo/ui/screens/countrylist/CountryListViewModel.kt

```diff
@@ -39,8 +64,10 @@ class CountryListViewModel(
         viewModelScope.launch {
             val message = try {
                 repository.fetchCountries()
+                _fetchCountriesSuccessful.value = true
                 "Successfully updated the list of countries"
             } catch (e: Exception) {
+                _fetchCountriesSuccessful.value = false
```

Comment:
Note: Rather than returning the boolean here through a new flow `fetchCountriesSuccessful` you could actually add a `try/catch` inside the repository's `fetchCountries()` method.  If the catch occurs, then pull the countries list from `countryDao`.  That way here, your try/catch won't get any exception, and instead will get this list from local storage if there was a network error.